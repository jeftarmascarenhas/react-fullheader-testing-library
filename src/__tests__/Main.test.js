import React from 'react'
import { render, cleanup } from '@testing-library/react'

import FullHeader from '../Main'

import videoSrc from '../assets/video/demo.mp4'

const renderIgnoringUnstableFlushDiscreteUpdates = component => {
  // tslint:disable: no-console
  const originalError = console.error
  const error = jest.fn()
  console.error = error
  const result = render(component)
  expect(error).toHaveBeenCalledTimes(1)
  expect(error).toHaveBeenCalledWith(
    'Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.%s',
    expect.any(String),
  )
  console.error = originalError
  // tslint:enable: no-console
  return result
}

afterEach(cleanup)

describe('<FullHeader />', () => {
  it('rendering', () => {
    const { asFragment } = render(<FullHeader />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('should have title when text to title passed', () => {
    const { getByText, queryByTestId } = render(
      <FullHeader title="TDD in React" />,
    )
    expect(getByText(/tdd in react/i).textContent).toEqual('TDD in React')
    expect(queryByTestId('title')).toBeDefined()
  })
  it('should not have title when title not passed', () => {
    const { queryByText } = render(<FullHeader />)
    expect(queryByText(/Curso TDD/i)).toBeNull()
  })
  it('should have subtitle when subtitle passed', () => {
    const subtitle = 'React Testing Librar'
    const { getByText } = render(<FullHeader subtitle={subtitle} />)
    expect(getByText(/React Testing Librar/i).textContent).toEqual(subtitle)
  })
  it('should have not subtitle when subtitle not passed', () => {
    const { queryByTestId } = render(<FullHeader />)
    expect(queryByTestId('subtitle')).toBeNull()
  })
  it('should have not text bgColor when bgColor passed', () => {
    const { queryByTestId } = render(<FullHeader bgColor="#eee" />)
    expect(queryByTestId('subtitle')).toBeNull()
  })
  it('should have not video when video tag passed', () => {
    const { queryByTestId } = render(<FullHeader bgColor="#eee" />)
    expect(queryByTestId('video')).toBeNull()
  })
  it('should have video when video tag', () => {
    const { queryByTestId } = renderIgnoringUnstableFlushDiscreteUpdates(
      <FullHeader video={videoSrc} autoPlay muted loop />,
    )
    const video = queryByTestId('video')
    expect(video).toBeTruthy()
  })
})
