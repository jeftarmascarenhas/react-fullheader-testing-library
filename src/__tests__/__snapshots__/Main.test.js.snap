// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullHeader /> rendering 1`] = `
<DocumentFragment>
  .c0 {
  height: 100vh;
  width: 100%;
  background-size: cover;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  background-image: url();
}

.c1 {
  position: relative;
  width: 100%;
  z-index: 2;
}

<header
    class="c0"
    data-testid="header"
  >
    <div
      class="c1"
    />
  </header>
</DocumentFragment>
`;
